bentoml/_internal/frameworks/transformers.py

```diff
@@ -106,427 +66,248 @@ def check_flax_supported() -> None:  # pragma: no cover
-        model_store (:mod:`~bentoml._internal.models.store.ModelStore`, default to :mod:`BentoMLContainer.model_store`):
-            BentoML modelstore, provided by DI Container.
-        from_tf (:code:`bool`, `optional`, defaults to :code:`False`):
-            Load the model weights from a TensorFlow checkpoint save file.
-        from_flax (:code:`bool`, `optional`, defaults to :code:`False`):
-            Load the model weights from a Flax checkpoint save file
-        return_config (:code:`bool`, `optional`, default to :code:`False`):
-            Whether or not to return configuration of the Transformers model.
-        config_kwargs (:code:`Dict[str, Any]`, `optional`):
-            Kwargs to pass into :code:`Config` object.
-        model_kwargs (:code:`Dict[str, Any]`, `optional`):
-            Kwargs to pass into :code:`Model` object.
-        tokenizer_kwargs (:code:`Dict[str, Any]`, `optional`):
-            Kwargs to pass into :code:`Tokenizer` object.
-        feature_extractor_kwargs (:code:`Dict[str, Any]`, `optional`):
-            Kwargs to pass into :code:`FeatureExtractor` object.
-        kwargs (:code:`Dict[str, Any]`, `optional`):
-            Other kwargs that can be parsed to transformers that is neither configs, model, tokenizer, and feature extractor.
-
-        .. warnings::
-            Make sure to add the corresponding kwargs for your Transformers :code:`Model`, :code:`Tokenizer`, :code:`Config`, :code:`FeatureExtractor` to the correct kwargs dict.
-
-        .. warnings::
-            Currently :code:`kwargs` accepts all kwargs for corresponding Pipeline.
+try:
+    import transformers
 
-    Returns:
-        :obj:`Union[Pipeline, Tuple[Optional[PretrainedConfig], Union[PreTrainedModel, TFPreTrainedModel, FlaxPreTrainedModel], Optional[Union[PreTrainedTokenizer, PreTrainedTokenizerFast, PreTrainedFeatureExtractor]]]]`: either returning a
-        pipeline or a tuple containing :obj:`PretrainedConfig`, :obj:`Model` class object defined by :obj:`transformers`, with an optional :obj:`Tokenizer` class, or :obj:`FeatureExtractor` class for the given model saved in BentoML modelstore.
+    _check_flax_supported()
+except ImportError:  # pragma: no cover
+    raise MissingDependencyException(
+        """\
+        transformers is required in order to use module `bentoml.transformers`.
+        Instruction: Install transformers with `pip install transformers`.
+        """
+    )
 
-    Examples:
 
-    .. code-block:: python
+@attr.define
+class TransformersOptions(ModelOptions):
+    """Options for the Transformers model."""
 
-        import bentoml
-        model, tokenizer = bentoml.transformers.load('custom_gpt2')
+    task: str = attr.field(
+        validator=[
+            attr.validators.instance_of(str),
+            lambda instance, attribute, value: transformers.pipelines.check_task(value),  # type: ignore
+        ]
+    )
 
-    If you want to returns an config object:
+    pipeline: bool = attr.field(
```

Comment:
where do we use this again? @ssheng

Reply:
We don't use it today. This is more of an informative metadata which is useful if we add support for models, tokenizers, config, and feature extractors. I'm also OK if we want to exclude this field for now.

Reply:
lets leave it out for now, since it doesn't provide any value yet. But agree with adding this when we have more metadata about model, tokenizer.